Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Driver.hs

```diff
@@ -1560,6 +1565,51 @@ getStats (driverId, _, merchantOpCityId) date = do
         bonusEarningWithCurrency = PriceAPIEntity bonusEarning currency
       }
 
+getEarnings :: (Id SP.Person, Id DM.Merchant, Id DMOC.MerchantOperatingCity) -> Day -> Day -> DEarning.EarningType -> Flow DEarning.EarningPeriodStatsRes
+getEarnings (driverId, _, merchantOpCityId) from to earningType = do
+  transporterConfig <- SCTC.findByMerchantOpCityId merchantOpCityId (Just (DriverId (cast driverId))) >>= fromMaybeM (TransporterConfigNotFound merchantOpCityId.getId)
+  case earningType of
+    DEarning.DAILY -> do
+      when (daysBetween /= transporterConfig.earningsWindowSize) $
+        throwError $ InvalidRequest "For daily earnings, the date range must be exactly 7 days (inclusive)."
```

Comment:
Why it can't be less than 7 days? And 7 should be configurable in your error message.
The same for weekly/monthly